### PR TITLE
feat: add ChatReplay

### PR DESCRIPTION
**Pull request by AnubisNekhet**
## 🎉 Theme for Website 🎉

[ChatReplay](https://chatreplay.stream) adds a twitch chat to VODs

## 💬 Additional Comments 💬

I added another param for `#catppuccin` called `platform` to apply separate styles to the chatbox and website.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/src/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `assets/mocha.webp`, `assets/macchiato.webp`, `assets/frappe.webp` and
    `assets/latte.webp` - individual screenshots of the themed website, in all 4
    Catppuccin flavors.
  - `catwalk.webp` - image of all individual screenshots stitched together,
    generated via [catwalk](https://github.com/catppuccin/toolbox#-catwalk).

### CODE REVIEW

File: styles/chatreplay/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        @base: @catppuccin[@@lookup][@base];
+        @mantle: @catppuccin[@@lookup][@mantle];
+        @crust: @catppuccin[@@lookup][@crust];
+        @accent: @catppuccin[@@lookup][@@accentColour];
+
+        & when (@lookup =latte) {
+        }
+
+        & when not (@lookup =latte) {
+        }
```
Comment (isabelroses) [inline styles/chatreplay/catppuccin.user.css:183]: If these are/will not be used then you can remove them

File: styles/chatreplay/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        @overlay2: @catppuccin[@@lookup][@overlay2];
+        @overlay1: @catppuccin[@@lookup][@overlay1];
+        @overlay0: @catppuccin[@@lookup][@overlay0];
+        @surface2: @catppuccin[@@lookup][@surface2];
+        @surface1: @catppuccin[@@lookup][@surface1];
+        @surface0: @catppuccin[@@lookup][@surface0];
+        @base: @catppuccin[@@lookup][@base];
+        @mantle: @catppuccin[@@lookup][@mantle];
+        @crust: @catppuccin[@@lookup][@crust];
+        @accent: @catppuccin[@@lookup][@@accentColour];
```
Comment (isabelroses) [inline styles/chatreplay/catppuccin.user.css:177]: These changes should be if you follow the change above, this ensures the least amount of changes that will be needed to make to the code
```suggestion
        @accent: @catppuccin[@@lookup][@accent-color];
```

File: styles/chatreplay/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+            @surface1: #45475a;
+            @surface0: #313244;
+            @base: #1e1e2e;
+            @mantle: #181825;
+            @crust: #11111b;
+        }
+    }
+
+
+    #catppuccin(@lookup, @accent, @platform) {
```
Comment (isabelroses) [inline styles/chatreplay/catppuccin.user.css:150]:
```suggestion
    #catppuccin(@lookup, @accent-color, @platform) {
```

**Review by isabelroses — CHANGES_REQUESTED** [review]
Thanks for your work, please see the suggested changes below.

<details>
<summary>Suggested changes</summary>

As you can see here the latte view for the chat settings is a bit hard to read.
image

The chat time stamps also appear to not be ctp themed
image

The background also does not appear to reach all the way to the bottom of the page
image

Can you also check [this](https://github.com/catppuccin/userstyles/actions/runs/5669319024/job/15361761934?pr=144) or the PR's diff to see all the typo warnings.

Can you also do this to make the sidebar match the center section or something similar
```diff
+++ #inner-a.tmgko.tmgko {
+++    background: @mantle
+++ }```

</details>

File: styles/chatreplay/catppuccin.user.css (hunk 1 above)
Comment (isabelroses) [reply styles/chatreplay/catppuccin.user.css]: > > The background also does not appear to reach all the way to the bottom of the page

> I've managed to fix every other issue except for this bug, I can't seem to replicate it

This was the [link](https://chatreplay.stream/videos/Pe-cE193YA8) i used if that helps

Comment (AnubisNekhet) [reply styles/chatreplay/catppuccin.user.css]: Fixed + Added options to change background opacity and blur parameters

File: styles/chatreplay/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+@version 0.0.1
+@updateURL https://github.com/catppuccin/userstyles/raw/main/styles/chatreplay/catppuccin.user.css
+@description Soothing pastel theme for ChatReplay
+
+@preprocessor   less
+@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
+@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent Color" [ "rosewater:Rosewater", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green*", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Grey" ]
+
+@var range bg-opacity "Background Opacity" [0.2 , -0.05 , 1 , 0.05]
```
[the remark this answers is not in the document]
Comment (AnubisNekhet) [reply styles/chatreplay/catppuccin.user.css:14]: Weird bug cause it simply reverts to default at 0

[the remark this answers is not in the document]
Comment (isabelroses) [reply styles/chatreplay/catppuccin.user.css:14]: Average stylus moment.
All good though

**Review by isabelroses — APPROVED** [review]

**Pull request merged**